src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java

```diff
@@ -831,20 +830,26 @@ private static boolean isVariableInOperatorExpr(
      */
     private static boolean isChild(DetailAST parent, DetailAST ast) {
         boolean isChild = false;
-        final ASTEnumeration astList = parent.findAllPartial(ast);
+        DetailAST curNode = parent.getFirstChild();
 
-        while (astList.hasMoreNodes()) {
-            final DetailAST astNode = (DetailAST) astList.nextNode();
-            DetailAST astParent = astNode.getParent();
+        while (curNode != null) {
+            if (curNode.equals(ast)) {
+                isChild = true;
+                break;
+            }
+
+            DetailAST toVisit = curNode.getFirstChild();
+            while (toVisit == null) {
+                curNode = curNode.getParent();
 
-            while (astParent != null) {
-                if (astParent.equals(parent)
-                        && astParent.getLineNo() == parent.getLineNo()) {
```

Comment:
This check is basically removed because we do `parent.getFirstChild()`.
The original method being used not only checked the children, but all the siblings as well. So the original method was checking for things outside the scope of what we are looking for.